File: listr-svc/ListR/ListR.Common/EntityModels/Users/UserGroup.cs
using ListR.DataLayer.EntityModels.Lists;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ListR.DataLayer.EntityModels.Users
{
    public class UserGroup
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? UserCreatedBy { get; set; }

        [JsonIgnore]
        public ICollection<User> Users { get; set; } = new List<User>();

        [JsonIgnore]
        public ICollection<ShopList> Lists { get; set; } = new List<ShopList>();

        [NotMapped]
        public List<string>? UserIds { get; set; }
        
        [NotMapped]
        public bool isCreator { get; set; }
    }
}

File: listr_app/ListR/lib/common/models/Data/user_group.dart
import 'package:listr/common/models/Data/shop_list.dart';
import 'package:listr/common/models/Data/user_model.dart';

class UserGroup {
  int? id;
  String? name;
  String? userCreatedBy;
  List<UserModel>? users;
  List<ShopList>? lists;
  List<String>? userIds;
  bool? isCreator;

  UserGroup(
      {this.id,
      this.name,
      this.userCreatedBy,
      this.users,
      this.lists,
      this.userIds,
      this.isCreator});

  UserGroup.fromJson(Map<String, dynamic> json){
    id = json['id'];
    name = json['name'];
    userCreatedBy = json['userCreatedBy'];
    userIds = json['userIds'];
    isCreator = json['isCreator'];

    lists = <ShopList>[];
    for (var element in (json['lists'] as List<dynamic>)) {
      lists?.add(ShopList.fromJson(element));
    }

    users = <UserModel>[];
    for (var element in (json['users'] as List<dynamic>)) {
      users?.add(UserModel.fromJson(element));
    }
  }



  Map<String, dynamic> toJson(){
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['userCreatedBy'] = userCreatedBy;
    data['users'] = users;
    data['lists'] = lists;
    data['userIds'] = userIds;
    data['isCreator'] = isCreator;
    return data;
  }
}